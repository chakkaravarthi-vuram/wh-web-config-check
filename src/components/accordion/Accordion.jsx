import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames/bind';
import ThemeContext from '../../hoc/ThemeContext';

import styles from './Accordion.module.scss';
import { ARIA_ROLES, BS } from '../../utils/UIConstants';
import gClasses from '../../scss/Typography.module.scss';
import { EMPTY_STRING } from '../../utils/strings/CommonStrings';
import ChevronUp from '../../assets/icons/ChevronUp';

function Accordion(props) {
  const { primaryColor } = useContext(ThemeContext);
  const {
    id,
    children,
    headerText,
    onHeaderClick,
    isChildrenVisible,
    className,
    headerClassName,
    hideBorder,
    childrenClassName,
    childrenStyle,
    chevronLabel,
    headerContent,
    headerContentClassName,
    onIconClickHandler,
    iconClassName,
    iconContainerClassName,
    iconRotationBeforeClassName,
    iconRotationAfterClassName,
    hideIcon,
  } = props;
  const borderClass = !hideBorder && styles.Border;
  return (
    <div id={id} className={cx(styles.Container, borderClass, className)}>
      <button
        className={cx(
          styles.Header,
          gClasses.CenterV,
          BS.JC_BETWEEN,
          gClasses.ClickableElement,
          BS.W100,
          gClasses.CursorPointer,
          headerClassName,
        )}
        onClick={onHeaderClick}
        tabIndex={onHeaderClick ? 0 : -1}
      >
        <div className={cx(gClasses.FTwo13, headerContentClassName)} style={{ color: primaryColor }}>
          {headerContent || headerText}
        </div>
        {!hideIcon &&
          <button
            className={cx(BS.D_FLEX, gClasses.CenterV, gClasses.ClickableElement, iconContainerClassName)}
            onClick={onIconClickHandler}
          >
            {chevronLabel && (
              <div className={cx(gClasses.FOne11BlackV11, gClasses.Italics, gClasses.MR15)}>
                {chevronLabel}
              </div>
            )}
            <ChevronUp
              role={ARIA_ROLES.IMG}
              ariaLabel={isChildrenVisible ? 'Collapse' : 'Expand'}
              className={cx(iconClassName, (isChildrenVisible ? iconRotationAfterClassName || null : iconRotationBeforeClassName || gClasses.Rotate180))}
            />
          </button>
        }
      </button>
      <div
        className={cx(childrenClassName, isChildrenVisible ? styles.Content : BS.D_NONE)}
        style={childrenStyle}
      >
        {children}
      </div>
    </div>
  );
}

export default Accordion;
Accordion.defaultProps = {
  children: null,
  isChildrenVisible: false,
  className: EMPTY_STRING,
  headerClassName: EMPTY_STRING,
  headerContent: null,
  headerContentClassName: EMPTY_STRING,
  iconClassName: EMPTY_STRING,
  iconContainerClassName: EMPTY_STRING,
  onIconClickHandler: () => {},
};
Accordion.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.string,
  ]),
  isChildrenVisible: PropTypes.bool,
  headerText: PropTypes.string.isRequired,
  onHeaderClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  headerClassName: PropTypes.string,
  headerContent: PropTypes.element,
  headerContentClassName: PropTypes.string,
  iconClassName: PropTypes.string,
  iconContainerClassName: PropTypes.string,
  onIconClickHandler: PropTypes.func,
};
