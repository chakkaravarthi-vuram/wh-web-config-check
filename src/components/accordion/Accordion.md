A graphical control element comprising a vertically stacked list of items. Example below

```js
<Accordion></Accordion>
```

```jsx static
import React from 'react';
<Accordion>{children}</Accordion>
```
