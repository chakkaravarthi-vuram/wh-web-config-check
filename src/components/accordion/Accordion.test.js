import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Accordion from './Accordion';

const props = {
  headerText:'test',
  onHeaderClick:()=>null
}

test('App component has been rendered correctly', () => {
  const wrapper = shallow(<Router><Accordion {...props} /></Router>);
  expect(wrapper).toMatchSnapshot();
});