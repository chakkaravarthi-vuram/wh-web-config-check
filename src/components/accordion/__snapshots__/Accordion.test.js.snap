// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component has been rendered correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Accordion
    className=""
    headerClassName=""
    headerText="test"
    isChildrenVisible={false}
    onHeaderClick={[Function]}
  />
</Router>
`;
